library/Vanilla/Formatting/Formats/HtmlFormat.php

```diff
@@ -77,6 +74,28 @@ public function __construct(
         $this->allowExtendedContent = $allowExtendedContent;
     }
 
+    /**
+     * Dependency injection.
+     *
+     * @param AttachmentHtmlProcessor $attachmentHtmlProcessor
+     * @param HeadingHtmlProcessor $headingHtmlProcessor
+     * @param ImageHtmlProcessor $imageHtmlProcessor
+     * @param UserContentCssProcessor $userContentCssProcessor
+
+     */
+    public function setDependencies(
+        AttachmentHtmlProcessor $attachmentHtmlProcessor,
+        HeadingHtmlProcessor $headingHtmlProcessor,
+        ImageHtmlProcessor $imageHtmlProcessor,
+        UserContentCssProcessor $userContentCssProcessor
+    ) {
+        $this->attachmentHtmlProcessor = $attachmentHtmlProcessor;
+        $this->headingHtmlProcessor = $headingHtmlProcessor;
+        $this->imageHtmlProcessor = $imageHtmlProcessor;
+        $this->addHtmlProcessor($headingHtmlProcessor);
+        $this->addHtmlProcessor($userContentCssProcessor);
```

Comment:
They are dependency injected, because these processors have specific methods used specifically in the HTML processor (for extracting images, headings, and attachments).

I could leave off this bit and configure some DI rules, I opted to use the ones I was already injecting because:

- They aren't necessary on the other formatters.
- They should always be enabled.
- They are already dependency injected.
- Since their always on, it seems easier to just add it here, rather than in our 3 bootstraps (app bootstrap, test bootstrap, minimal/mocked test bootstrap).